### ISSUE TITLE
MODX 3 alpha1 - icons visable not actions

### ISSUE THREAD

**Issue by OlegShchavelev** [modxcms/revolution#14929]
## Bug report

The navigation icon is displayed if there is no context menu.

![изображение](https://user-images.githubusercontent.com/20732384/71304954-5391d700-23de-11ea-90ab-49b6ce6a1e81.png)

### Environment
MODX version, apache/nginx and version, mysql version, browser, etc. Any relevant information.

**Comment by Ruslan-Aleev** [comment]
> The list of package versions does have a context menu - only the most recent version listed does not. So removing the icon is not a proper fix here.

@Mark-H Perhaps we can close this issue? Because there is, in fact, no bug.

Although the components, it seems to me, are more often updated than not and maybe it’s worth hiding the gear-icon? Not sure.

**Comment by JoshuaLuckers** [comment]
We have to investigate whether it is possible to hide/show the gear-icon per row and maybe even what "actions" it shows. For packages that don't have any actions we should hide the gear-icon.

**Comment by Ibochkarev** [comment]
The designated problem remains relevant. Tested on the latest version MODX Revolution 3.0.0-alpha3 (git)

![image](https://user-images.githubusercontent.com/2138260/93712996-0c0dc680-fb7b-11ea-8fbb-2be2bade72b7.png)

**Comment by smg6511** [comment]
This is still quite relevant and there are actually issues in quite a few places, in terms of menu options, buttons, and the context and actions menus not accurately reflecting what one has permissions to do. I've been working on this over the past few days now and will submit a PR in the near future.

@JoshuaLuckers - I've figured out how to handle the actions (gear icon) menu visibility on a per-row basis. But in addition to that issue, none of the other avenues of editing/creating (the column-level editors, the create buttons, the batch [delete] buttons, etc.) are in sync with the permissions level a user has. So I'm sifting through that. 

The real tricky thing is in the Roles grid: We have two (what I'm calling) protected options (_Super User_ and _Member_) that should never be editable, whereas the editability of rest of the grid's options is permissions-dependent. There's no built-in way to enable/disable the editor(s) on a row-by-row basis, so that's proving to require a creative solution—one that's in progress.

Anyway, I do have a question for those who might review this: I think it'd be much more user-friendly to have the column editor activate on one click instead of two. To me, when one hovers over a cell and see's the pencil icon, they're expecting something to happen when they click. It's a simple config change, not hard to do. @Ruslan-Aleev - I'd like to get your opinion on this as well, as I know you're very UI-focused.

**Comment by Ruslan-Aleev** [comment]
> Anyway, I do have a question for those who might review this: I think it'd be much more user-friendly to have the column editor activate on one click instead of two.

Yes, the number of clicks is confusing, we discussed this in PR https://github.com/modxcms/revolution/pull/14748#issuecomment-534146728
But with the current implementation, this cannot be quickly fixed, since the pencil icon is only a visual indication that the grid fields are editable. If we make the fields editable by 1 click, then users will often activate the editing by mistake. In general, ideally, we need to make the icon a separate button, and not just an indication.

**Comment by smg6511** [comment]
> If we make the fields editable by 1 click, then users will often activate the editing by mistake.

The funny thing is, for me because the pen icon appears, I expect to be able to edit upon one click (anywhere on the column and not necessarily a click on the pen icon itself). It'd be interesting to know what others expect too. I'm learning that we have very different expectations on this particular UI element (icons that appear as a visual cue on hover)—I don't expect the icons to be the trigger (unless another visual cue occurs when hovering over that icon). In the case of these editable columns, perhaps the icon is just overkill as we already provide the cue that the field can be manipulated by changing the cursor on hover. The thing we can probably agree on here is that there's likely a learning curve of a handful of clicks for either strategy to know its behavior (the current one I'm very familiar with as I've used MODX since the pre-Revo days)—I guess it's just occurred to me while working on this issue that it really is annoying (to me) to have to click twice to edit.

@Mark-H - It'd be good to get your take on this as well (I see the PR that @Ruslan-Aleev refers to was yours and relates to exactly what I'm asking about).

**Comment by Mark-H** [comment]
Cases like this I'll happily defer to others with more practical UX knowledge to reach consensus. I can see benefits to each approach of single or double click. 

Primary reason it is double click now is because that's how the grid worked before. The PR only added some emphasis and the pen icon to indicate what fields are in fact editable.

**Comment by smg6511** [comment]
Considering there may not be strong opinion toward one way or the other, perhaps we put it in the user's control and add a system setting to choose single or double click. That could appease individual preferences, and we just keep the default behavior as-is.

**Comment by Ruslan-Aleev** [comment]
@smg6511 In general, it is worth making a separate issue, since the discussion of the pencil icon does not apply to this issue of context menu icons. The current pencil icon improves UX (previously it was not clear that the field in the grid could be edited), but there are always better solution :)

**Comment by smg6511** [comment]
@Ruslan-Aleev True, I'll leave this question to another possible PR. Maybe I'll propose the single/double-click setting in a new issue (as a feature). FWIW, I don't dislike the use of the pencil icon in general, I'm just not crazy about getting that cue, the cursor cue, _and_ having to double-click to edit a given field. ;-)

**Comment by rthrash** [comment]
This issue has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/need-insight-on-acl-implementation-in-back-end/4623/1

**Comment by smg6511** [comment]
This issue will be fixed when #16653 gets finalized and merged. Stay tuned...
